packages/now-routing-utils/src/superstatic.ts

```diff
@@ -197,13 +186,25 @@ function replaceSegments(
       hash,
     });
 
-    // url.format() escapes the query string but we must preserve dollar signs
-    destination = destination.replace(/=%24/g, '=$');
+    // url.format() escapes the dollar sign but it must be preserved for now-proxy
+    destination = destination.replace(/%24/g, '$');
   }
 
   return destination;
 }
 
+function safelyCompile(str: string, indexes: { [k: string]: string }): string {
+  if (!str) {
+    return str;
+  }
+  // path-to-regexp cannot compile question marks
+  return str
+    .split('?')
+    .map(part => compile(part))
+    .map(func => func(indexes))
```

Comment:
```suggestion
    .map(part => compile(part)(indexes))
```
Small nit, we could combine these into one loop

Reply:
I changed to use one `.map()`. I left it as too lines so we can see which one failed: creating the compiler or invoking the compiler.